plugins/providers/bigquery/provider_test.go

```diff
@@ -0,0 +1,1110 @@
+
+		actualError := p.CreateConfig(pc)
+
+		assert.NoError(t, actualError)
+	})
+}
+
+func TestGetResources(t *testing.T) {
+	t.Run("should error when credentials are invalid", func(t *testing.T) {
+		crypto := new(mocks.Crypto)
+		p := bigquery.NewProvider("", crypto)
+		pc := &domain.ProviderConfig{
+			Type:        domain.ProviderTypeBigQuery,
+			URN:         "test-project-id",
+			Credentials: "invalid-creds",
+		}
+
+		actualResources, actualError := p.GetResources(pc)
+
+		assert.Nil(t, actualResources)
+		assert.Error(t, actualError)
+	})
+
+	t.Run("should return dataset resource object", func(t *testing.T) {
+		crypto := new(mocks.Crypto)
+		client := new(mocks.BigQueryClient)
+		p := bigquery.NewProvider("", crypto)
+		p.Clients = map[string]bigquery.BigQueryClient{
+			"resource-name": client,
+		}
+		validCredentials := bigquery.Credentials{
+			ServiceAccountKey: base64.StdEncoding.EncodeToString([]byte("service-account-key-json")),
+			ResourceName:      "projects/resource-name",
+		}
+		pc := &domain.ProviderConfig{
+			Type:        domain.ProviderTypeBigQuery,
+			URN:         "test-project-id",
+			Credentials: validCredentials,
+			Resources: []*domain.ResourceConfig{
+				{
+					Type: "dataset",
+					Roles: []*domain.Role{
+						{
+							ID:          "VIEWER",
+							Name:        "VIEWER",
+							Permissions: []interface{}{"VIEWER"},
+						},
+					},
+				},
+				{
+					Type: "table",
+					Roles: []*domain.Role{
+						{
+							ID:          "VIEWER",
+							Name:        "VIEWER",
+							Permissions: []interface{}{"VIEWER"},
+						},
+					},
+				},
+			},
+		}
+		expectedDatasets := []*bigquery.Dataset{
+			{
+				ProjectID: "p_id",
+				DatasetID: "d_id",
+			},
+		}
+		expectedTables := []*bigquery.Table{
+			{
+				ProjectID: "p_id",
+				DatasetID: "d_id",
+				TableID:   "t_id",
+			},
+		}
+		//		crypto.On("Decrypt", "c2VydmljZS1hY2NvdW50LWtleS1qc29u").Return(`{"type":"service_account"}`, nil).Once()
+		client.On("GetDatasets", mock.Anything).Return(expectedDatasets, nil).Once()
+		client.On("GetTables", mock.Anything, mock.Anything).Return(expectedTables, nil).Once()
+		expectedResources := []*domain.Resource{
+			{
+				ProviderType: domain.ProviderTypeBigQuery,
+				ProviderURN:  "test-project-id",
+				Type:         "dataset",
+				Name:         "d_id",
+				URN:          "p_id:d_id",
+			},
+			{
+				ProviderType: domain.ProviderTypeBigQuery,
+				ProviderURN:  "test-project-id",
+				Name:         "t_id",
+				URN:          "p_id:d_id.t_id",
+				Type:         "table",
+			},
+		}
+		actualResources, actualError := p.GetResources(pc)
+
+		assert.Equal(t, expectedResources, actualResources)
+		assert.Nil(t, actualError)
+	})
+}
+
+func TestGrantAccess(t *testing.T) {
+
+	t.Run("should return error if Provider Config or Appeal doesn't have required paramters", func(t *testing.T) {
+		testCases := []struct {
+			providerConfig *domain.ProviderConfig
```

Comment:
```suggestion
		testCases := []struct {
		    name           string
			providerConfig *domain.ProviderConfig
```
can we add `name` to describe each test cases?